daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala

```diff
@@ -21,101 +23,127 @@ class DarReader[A](
     // The `Long` is the dalf size in bytes.
     parseDalf: (Long, InputStream) => Try[A]) {
 
-  import Errors._
+  import DarReader._
 
-  private val manifestEntry = new ZipEntry("META-INF/MANIFEST.MF")
+  def readArchiveFromFile(darFile: File) =
+    readArchive(darFile.getName, new ZipInputStream(new FileInputStream(darFile)))
```

Comment:
@gaboraranyossy-da we are leaking the `ZipInputStream` create here. It should be closed.

Reply:
good spot! it's addressed now.